tools/check_format.py

```diff
@@ -68,13 +68,36 @@ def checkProtobufExternalDepsBuild(file_path):
 def checkProtobufExternalDeps(file_path):
   if whitelistedForProtobufDeps(file_path):
     return True
+  protobuf_type_errors = {
+      # Well-known types should be referenced from the ProtobufWkt namespace.
+      "Protobuf::Any":                    "ProtobufWkt::Any",
+      "Protobuf::Empty":                  "ProtobufWkt::Empty",
+      "Protobuf::ListValue":              "ProtobufWkt:ListValue",
+      "Protobuf::NULL_VALUE":             "ProtobufWkt::NULL_VALUE",
+      "Protobuf::StringValue":            "ProtobufWkt::StringValue",
+      "Protobuf::Struct":                 "ProtobufWkt::Struct",
+      "Protobuf::Value":                  "ProtobufWkt::Value",
+
+      # Maps including strings should use the protobuf string types.
+      "Protobuf::MapPair<std::string":    "Protobuf::MapPair<Envoy::ProtobufTypes::String",
+
+      # Other common mis-namespacing of protobuf types.
+      "ProtobufWkt::Map":                 "Protobuf::Map",
+      "ProtobufWkt::MapPair":             "Protobuf::MapPair",
+      "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
+  }
+
   with open(file_path) as f:
     text = f.read()
     if '"google/protobuf' in text or "google::protobuf" in text:
       printError(
           "%s has unexpected direct dependency on google.protobuf, use "
           "the definitions in common/protobuf/protobuf.h instead." % file_path)
       return False
+    for error, replacement in protobuf_type_errors.items():
```

Comment:
I don't feel that strongly about it, but can we do all of this on a per-line basis with all the other checks? The format script is getting pretty slow.

Reply:
Moved this to fixFileContents since the fixes can be performed automatically. I'm going to do a little digging and see if I can speed up the rest of the script.

Reply:
Opened #2974 to parallelize formatting. From profiling, it looks like the majority of the time is still spent running clang-format, with header order checking being the next most time-consuming.